src/internal.c

```diff
@@ -2202,6 +2208,13 @@ void SSL_ResourceFree(WOLFSSL* ssl)
 #if defined(KEEP_PEER_CERT) || defined(GOAHEAD_WS)
     FreeX509(&ssl->peerCert);
 #endif
+#ifdef HAVE_SESSION_TICKET
+    if (ssl->session.isDynamic) {
+        XFREE(ssl->session.ticket, ssl->heap, DYNAMIC_TYPE_SESSION_TICK);
```

Comment:
Shouldn't the dynamic ticket be released even earlier, in FreeHandshakeResources() ?  And it still needs to be freed here in case FreeHandshakeResources() doesn't get called.